src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PENamedTypeSymbol.cs

```diff
@@ -2336,25 +2379,61 @@ private IEnumerable<PENamedTypeSymbol> CreateNestedTypes()
             return map;
         }
 
-        private void CreateProperties(Dictionary<MethodDefinitionHandle, PEMethodSymbol> methodHandleToSymbol, ArrayBuilder<Symbol> members)
+        internal static bool IsSynthesizedExtensionThisParameter(TypeSymbol extendedType, ParameterSymbol parameter)
+        {
+            return parameter.Type.Equals(extendedType, TypeCompareKind.AllIgnoreOptions) &&
+                   parameter.TypeWithAnnotations.CustomModifiers.All(static (modifier) => modifier.IsOptional) &&
+                   parameter.TypeWithAnnotations.CustomModifiers.Any(
+                       static (modifier) => SourceAttributeData.IsTargetAttribute(
+                                                ((CSharpCustomModifier)modifier).ModifierSymbol,
+                                                AttributeDescription.CaseSensitiveExtensionAttribute.Namespace,
+                                                AttributeDescription.CaseSensitiveExtensionAttribute.Name));
+        }
+
+        private void CreateProperties(Dictionary<MethodDefinitionHandle, (PEMethodSymbol, PEExtensionInstanceMethodSymbol)> methodHandleToSymbol, ArrayBuilder<Symbol> members)
         {
             var moduleSymbol = this.ContainingPEModule;
             var module = moduleSymbol.Module;
 
             try
             {
+                TypeSymbol extendedType = GetDeclaredExtensionUnderlyingType();
+
                 foreach (var propertyDef in module.GetPropertiesOfTypeOrThrow(_handle))
                 {
                     try
                     {
                         var methods = module.GetPropertyMethodsOrThrow(propertyDef);
 
-                        PEMethodSymbol getMethod = GetAccessorMethod(module, methodHandleToSymbol, _handle, methods.Getter);
-                        PEMethodSymbol setMethod = GetAccessorMethod(module, methodHandleToSymbol, _handle, methods.Setter);
+                        (PEMethodSymbol getMethod, PEExtensionInstanceMethodSymbol extensionGet) = GetAccessorMethod(module, methodHandleToSymbol, _handle, methods.Getter);
+                        (PEMethodSymbol setMethod, PEExtensionInstanceMethodSymbol extensionSet) = GetAccessorMethod(module, methodHandleToSymbol, _handle, methods.Setter);
 
                         if (((object)getMethod != null) || ((object)setMethod != null))
                         {
-                            members.Add(PEPropertySymbol.Create(moduleSymbol, this, propertyDef, getMethod, setMethod));
+                            PEPropertySymbol property = PEPropertySymbol.Create(moduleSymbol, this, propertyDef, getMethod, extensionGet, setMethod, extensionSet);
+
+                            // If this check causes a perf problem, we could optimize it.
+                            if (extendedType is not null && property.IsStatic && property.ParameterCount > 0 &&
+                                IsSynthesizedExtensionThisParameter(extendedType, property.Parameters[0]))
```

Comment:
Should we also check `extensionGet is not null || extensionSet is not null` like we do for events?

Reply:
> Should we also check extensionGet is not null || extensionSet is not null like we do for events?

Unlike an event, a property has its own signature, which by itself defines if we need to transform it.